noxfile.py

```diff
@@ -73,8 +73,10 @@ def run_examples(session):
 
 @nox.session(name="dev", reuse_venv=True)
 def set_dev(session):
-    set_environment_variables(session=session)
+    homedir = os.getenv("HOME")
+    LD_LIBRARY_PATH = f"{homedir}/.local/lib:{session.env.get('LD_LIBRARY_PATH')}"
     envbindir = session.bin
+    session.install("-e", ".[dev]")
     session.install("cmake")
     session.run(
```

Comment:
Does this session really require the `run` line? I think the env variables can just be set using `set_environment_variables(session=session)`?

Reply:
I believe it is required as we are using `bash` to export variables in the shell.

Reply:
But we aren't doing this in other sessions?

Reply:
We are't doing it other sessions because we don't have to do `source .nox/session/bin/activate` with any other session.
But in case to dev it is required to activate the dev env further.

Reply:
Makes sense, but we should not redefine the variables in `set_dev`, instead, we can read them from the environment and pass those values in the `run` command